base/bucket_gocb.go

```diff
@@ -104,7 +105,20 @@ func GetCouchbaseBucketGoCB(spec BucketSpec) (bucket *CouchbaseBucketGoCB, err e
 		EnableGoCBLogging()
 	}
 
-	cluster, err := gocb.Connect(spec.Server)
+	connSpec, err := gocbconnstr.Parse(spec.Server)
+	if err != nil {
+		return nil, err
+	}
+
+	// Increase the number of idle connections per-host to fix SG #3534
+	if connSpec.Options == nil {
+		connSpec.Options = map[string][]string{}
+	}
+	connSpec.Options["http_max_idle_conns_per_host"] = []string{fmt.Sprintf("%d", DefaultHttpMaxIdleConnsPerHost)}
+	connSpec.Options["http_max_idle_conns"] = []string{fmt.Sprintf("%d", DefaultHttpMaxIdleConns)}
+	connSpec.Options["http_idle_conn_timeout"] = []string{fmt.Sprintf("%d", DefaultHttpIdleConnTimeoutMilliseconds)}
```

Comment:
What do you think about casting to url.Values here, to use url.Values.Add() instead of the somewhat ugly string gymnastics above?  Something like:

```
        asValues := url.Values(options)
	asValues.Add("http_max_idle_conns_per_host", DefaultHttpMaxIdleConnsPerHost)
	connSpec.Options = asValues
```

Reply:
Good call, this is nicer looking